cabal-testsuite/src/Test/Cabal/Prelude.hs

```diff
@@ -507,11 +507,20 @@ withRepo repo_dir m = do
     -- 2. Create tarballs
     pkgs <- liftIO $ getDirectoryContents (testCurrentDir env </> repo_dir)
     forM_ pkgs $ \pkg -> do
+        let srcPath = testCurrentDir env </> repo_dir </> pkg
+        let destPath = package_dir </> pkg
+        isPreferredVersionsFile <- liftIO $
+            if pkg == "preferred-versions"
+                then doesFileExist srcPath
```

Comment:
does this check that it's not a directory? Unclear to me.

Reply:
indeed yes

Reply:
My thought process was bascially to have an additional check whether this is the correct file.

Reply:
Actually, I think the blocker may be @phadej's review status set to 'request changes'. The review would need to be finished by Oleg or dismissed by an admin (e.g., me) to unblock this. Nevertheless, let me run CI again before asking Oleg for help.

Reply:
Resolving didn't work, so on @phadej's request I 'dismissed' the review and restarted the CI. Let's see...